ql/src/java/org/apache/hadoop/hive/ql/parse/MergeSemanticAnalyzer.java

```diff
@@ -350,26 +352,51 @@ private MergeStatement.InsertClause handleInsert(ASTNode whenNotMatchedClause, S
       throw new SemanticException(String.format("Column schema must have the same length as values (%d vs %d)",
           columnListNode.getChildCount(), valuesNode.getChildCount() - 1));
     }
-    UnparseTranslator defaultValuesTranslator = new UnparseTranslator(conf);
-    defaultValuesTranslator.enable();
-    List<String> targetSchema = processTableColumnNames(columnListNode, targetTable.getFullyQualifiedName());
-    collectDefaultValues(valuesNode, targetTable, targetSchema, defaultValuesTranslator);
-    defaultValuesTranslator.applyTranslations(ctx.getTokenRewriteStream());
-    String valuesClause = getMatchedText(valuesNode);
-    valuesClause = valuesClause.substring(1, valuesClause.length() - 1); //strip '(' and ')'
 
-    String extraPredicate = getWhenClausePredicate(whenNotMatchedClause);
-    return new MergeStatement.InsertClause(
-        getMatchedText(columnListNode), valuesClause, onClausePredicate, extraPredicate);
-  }
+    List<String> columnNames;
+    if (columnListNode != null) {
```

Comment:
can it be null, there is no check above for null - columnListNode.getChildCount()?

Reply:
refactored this part